src/BloomExe/WebLibraryIntegration/BloomS3Client.cs

```diff
@@ -398,21 +440,47 @@ public string DownloadBook(string storageKeyOfBookFolder, string pathToDestinati
 				var destinationPath = Path.Combine(pathToDestinationParentDirectory, Path.GetFileName(children[0]));
 
 				//clear out anything exisitng on our target
+				var didDelete = false;
 				if (Directory.Exists(destinationPath))
 				{
-					Directory.Delete(destinationPath, true);
+					try
+					{
+						Directory.Delete(destinationPath, true);
+						didDelete = true;
+					}
+					catch (IOException)
+					{
+						// can't delete it...see if we can copy into it.
+					}
 				}
 
 				//if we're on the same volume, we can just move it. Else copy it.
 				// It's important that books appear as nearly complete as possible, because a file watcher will very soon add the new
 				// book to the list of downloaded books the user can make new ones from, once it appears in the target directory.
-				if (PathUtilities.PathsAreOnSameVolume(pathToDestinationParentDirectory, tempDestination.FolderPath))
+				bool done = false;
+				if (didDelete && PathUtilities.PathsAreOnSameVolume(pathToDestinationParentDirectory, tempDestination.FolderPath))
 				{
-					Directory.Move(children[0], destinationPath);
+					try
+					{
+						Directory.Move(children[0], destinationPath);
+						done = true;
+					}
+					catch (IOException)
+					{
+						// If moving didn't work we'll just try copying
+					}
+					catch (UnauthorizedAccessException)
+					{ }
+
 				}
-				else
+				if (!done)
+					done = CopyDirectory(children[0], destinationPath);
+				if (!done)
 				{
-					CopyDirectory(children[0], destinationPath);
+					var msg = LocalizationManager.GetString("Download.CopyFailed",
+						"Bloom downloaded your book but had problems copying it to the bloom library. Possibly you already have a version of this book and some program is using it. You may need to restart your computer and download it again.");
```

Comment:
"your" --> "the"
"copying it to the bloom library" --> "making it available in Bloom" 
"Possibly..." -> null
"You may need" --> "Please restart your computer and try again. If you get this message again, please write to us at issues@bloomlibrary.org".